### PR TITLE
Changed `azurecr` references to `ghcr`

### PR DESCRIPTION
**Pull request by jasonviviano**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.io/community/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Tested out packages by running the following command: `docker pull ghcr.io/.....`.

Following packages are still missing possibly:

-  radius.azurecr.io/helm/v1/repo
-  radius.azurecr.io/appcore-rp

Pulling the `ghcr.io/radius-project/recipes/local-dev/` packages all result in the following output:
`unable to fetch descriptor (sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855) which reports content size of zero: invalid argument`

### Summary
🚚📝🆕

This pull request updates the documentation and code snippets to use the new public registry `radius.ghcr.io` instead of the old one `radius.azurecr.io`. This change affects the guides for installing Radius on Kubernetes, authoring and registering custom recipes, authoring apps with various resources and networking options, and deploying apps with environments.

> _Oh, we're the crew of the `radius` ship_
> _And we sail the cloud with skill and wit_
> _We've changed our registry from `azurecr` to `ghcr`_
> _So pull the ropes and update the docs, me hearties!_

### Walkthrough
*  Update image URLs for demo app, AWS app, and volumes app from `radius.azurecr.io` to `radius.ghcr.io` in various code snippets and guides ([link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-16c6b1f038bfac9806df34f162b60508f10bfe6eb84a7b62f2a1d7c8d83fd23cL76-R76), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-e2dfb4c0df60a1e799ec3f30aaeb32b23366c6bfd35e3ffc472a31cb0f85b41fL13-R13), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-56ad111af2e11d8e414271b3b761ff7e867d56eed97617970bf53e8678da40a9L12-R12), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-be48532816f46e63784fa55a4754dad4d706abbef6f41398725eccd7a9444d4fL12-R12), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-6bafe9c1e3a55a85bbfc9466d9bfc98d00813be3a294349a487cbbcf3993ca6cL48-R48), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-c172da33cbf23fe678988b1e7f83ff8779a4d922b990cde64d9b5a28779b065dL49-R49), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-e53cad9ad9d60e11d1d4d50a7437c8589cdc7036957f1f85db6398db9cf86801L44-R44), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-7e5cc81d57656a3cc50877d40b991eeedb36d1aa9d5f1f6289cc4bba0c1c7df5L13-R13), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-6dd0a3940091e1b55fb7789f22226c421b0e1148a97874c0d1c03d3da6f34251L11-R11), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-c90092fc23218ef96c5fad9e2e574d25cf2d945fc495d5d922b289750b4eef21L17-R17), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-db4f42d91ba63f45581f45e1edbdc6e16e2e7cc692332291a7b9b06e69ff39edL18-R18), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-f514c6bf6a78ae5077ea611cd856ccc68bd719db2aba506197b56f2072b4dd3fL12-R12), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-59d4c4282c887953a6d82bc8070d4a198e25ef3c409bcfdef4c6c3cdc2a4199fL11-R11), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-917d43aa773e02df5bfc677a66ad5af9ffe844cf8728c00496e43be466ebe74aL12-R12), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-bba65600a8a7e5226e47aa0e73c9646ecccb9f7d02cd0e0d296e2563770186daL12-R12), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-ea6a9930a1177afce126cad63cd945f466de272f0796ee8f0b3816bb19e94fb1L12-R12))
* Update image URL for Azure workload identity app from `ghcr.io` to `radius.ghcr.io` in `container-wi.bicep` code snippet ([link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-de8e6eb6e7a98ae9ef00d19e6e88eb12aff02a1f0431e15b5d0380d7c7cf18a6L44-R44))
* Update template paths for MongoDB and Redis recipes from `radius.azurecr.io` to `radius.ghcr.io` in `rad recipe list` output ([link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-e84436cecc84636c9b6f2ecd9f6704a1bbe4fc0ffbb407deb51181ae5c26b1a3L69-R70))
* Update Helm repository URL from `radius.azurecr.io` to `radius.ghcr.io` in Kubernetes installation instructions ([link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-5b1e8b14bb206354b9eccd6444ecf75fc7c08aa75b63c229da510b4ea8ff1345L30-R30))
* Update image URLs for Radius resource provider, deployment engine, and universal control plane from `radius.azurecr.io` to `radius.ghcr.io` in Helm configuration options ([link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-5b1e8b14bb206354b9eccd6444ecf75fc7c08aa75b63c229da510b4ea8ff1345L50-R55))
* Update target path and template path for custom recipe from `myregistry.azurecr.io` to `myregistry.ghcr.io` in recipe authoring instructions ([link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-98e11a5783b838bbd05527cc9c92c645024b0d68921e043dc9cad5439cff01cbL110-R110), [link](https://github.com/radius-project/docs/pull/893/files?diff=unified&w=0#diff-98e11a5783b838bbd05527cc9c92c645024b0d68921e043dc9cad5439cff01cbL132-R132))

## Issue reference
https://github.com/radius-project/docs/issues/889

### CODE REVIEW

**Comment by AaronCrawfis** [comment]
Thanks for finding these! Our new registry is `ghcr.io/radius-project` instead of `radius.ghcr.io`. Could you update this PR to use `ghcr.io/radius-project` please?

File: docs/content/guides/recipes/howto-author-recipes/index.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ Resources are populated automatically for Bicep Recipes for any new Azure or AWS
 Recipes leverage [Bicep registries](https://learn.microsoft.com/azure/azure-resource-manager/bicep/private-module-registry) for template storage. Once you've authored a Recipe, you can publish it to your preferred OCI-compliant registry with [`rad bicep publish`]({{< ref rad_bicep_publish >}}):
 
 ```bash
-rad bicep publish --file myrecipe.bicep --target br:myregistry.azurecr.io/recipes/myrecipe:1.1.0
+rad bicep publish --file myrecipe.bicep --target br:myregistry.ghcr.io/recipes/myrecipe:1.1.0
```
Comment (jasonviviano) [inline docs/content/guides/recipes/howto-author-recipes/index.md]: Need help checking that the target path for the command would still technically be valid here or if this requires `azurecr`.

File: docs/content/guides/recipes/howto-author-recipes/index.md [hunk 2]
```diff
@@ -129,7 +129,7 @@ Now that your Recipe template has been stored, you can add it your Radius Enviro
 {{% codetab %}}
 
 ```bash
-rad recipe register myrecipe --environment myenv --resource-type Applications.Datastores/redisCaches --template-kind bicep --template-path myregistry.azurecr.io/recipes/myrecipe:1.1.0
+rad recipe register myrecipe --environment myenv --resource-type Applications.Datastores/redisCaches --template-kind bicep --template-path myregistry.ghcr.io/recipes/myrecipe:1.1.0
```
Comment (jasonviviano) [inline docs/content/guides/recipes/howto-author-recipes/index.md]: Need help checking that the target path for the command would still technically be valid here or if this requires `azurecr`.

File: docs/content/guides/recipes/howto-author-recipes/snippets/environment.bicep [hunk 3]
```diff
@@ -12,7 +12,7 @@ resource env 'Applications.Core/environments@2023-10-01-preview' = {
       'Applications.Datastores/redisCaches':{
         'redis-bicep': {
           templateKind: 'bicep'
-          templatePath: 'https://myregistry.azurecr.io/recipes/myrecipe:1.1.0'
+          templatePath: 'https://myregistry.ghcr.io/recipes/myrecipe:1.1.0'
```
Comment (jasonviviano) [inline docs/content/guides/recipes/howto-author-recipes/snippets/environment.bicep]: Need help checking that the target path for the command would still technically be valid here or if this requires `azurecr`.

File: docs/content/reference/resource-schema/core-schema/container-schema/_index.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ weight: 300
 
 | Key  | Required | Description | Example |
 |------|:--------:|-------------|---------|
-| image | y | The registry and image to download and run in your container. Follows the format `<registry-hostname>:<port>/<image-name>:<tag>` where registry hostname is optional and defaults to the Docker public registry, port is optional and defaults to 443, tag is optional and defaults to `latest`.| `myregistry.azurecr.io/myimage:latest`
+| image | y | The registry and image to download and run in your container. Follows the format `<registry-hostname>:<port>/<image-name>:<tag>` where registry hostname is optional and defaults to the Docker public registry, port is optional and defaults to 443, tag is optional and defaults to `latest`.| `myregistry.ghcr.io/myimage:latest`
```
Comment (jasonviviano) [inline docs/content/reference/resource-schema/core-schema/container-schema/_index.md]: Need help checking that the target path for the command would still technically be valid here or if this requires `azurecr`.

File: docs/content/reference/resource-schema/core-schema/container-schema/snippets/base-container.yaml [hunk 5]
```diff
@@ -16,4 +16,4 @@ spec:
     spec:
       containers:
         - name: frontend
-          image: radiusdev.azurecr.io/magpie:latest
\ No newline at end of file
+          image: radiusdev.ghcr.io/magpie:latest
```
Comment (jasonviviano) [inline docs/content/reference/resource-schema/core-schema/container-schema/snippets/base-container.yaml]: Need help identifying if this package called `magpie` maps to `magpieo`, `ghcr.io/radius-project/magpiego:latest`.

File: docs/content/reference/resource-schema/core-schema/environment-schema/index.md [hunk 6]
```diff
@@ -56,7 +56,7 @@ Details on what to run and how to run it are defined in the `container` property
 |------|:--------:|-------------|---------|
 | \<recipe-name\> | y | The name of the Recipe. Must be unique within the resource-type. | `myrecipe`
 | templateKind | y | Format of the template provided by the recipe. Allowed values: bicep | `'bicep'`
-| templatePath | y | The path to the Recipe contents. For Bicep Recipes this is a Bicep module registry address. | `'mycr.azurecr.io/recipes/myrecipe:1.0'`
+| templatePath | y | The path to the Recipe contents. For Bicep Recipes this is a Bicep module registry address. | `'mycr.ghcr.io/recipes/myrecipe:1.0'`
```
Comment (jasonviviano) [inline docs/content/reference/resource-schema/core-schema/environment-schema/index.md]: Need help, is this still a valid path example?

File: docs/content/reference/resource-schema/core-schema/extender/snippets/extender-manual.bicep [hunk 7]
```diff
@@ -30,7 +30,7 @@ resource publisher 'Applications.Core/containers@2023-10-01-preview' = {
   properties: {
     application: app.id
     container: {
-      image: 'radius.azurecr.io/magpie:latest'
+      image: 'radius.ghcr.io/magpie:latest'
```
Comment (jasonviviano) [inline docs/content/reference/resource-schema/core-schema/extender/snippets/extender-manual.bicep]: TODO: See above, is this mapping to the `magpieo` package I mentioned above?

File: docs/content/reference/resource-schema/core-schema/extender/snippets/extender-recipe.bicep [hunk 8]
```diff
@@ -21,7 +21,7 @@ resource publisher 'Applications.Core/containers@2023-10-01-preview' = {
   properties: {
     application: application
     container: {
-      image: 'radius.azurecr.io/magpie:latest'
+      image: 'radius.ghcr.io/magpie:latest'
```
Comment (jasonviviano) [inline docs/content/reference/resource-schema/core-schema/extender/snippets/extender-recipe.bicep]: TODO: Same as here

File: docs/content/reference/resource-schema/dapr-schema/dapr-pubsub/snippets/dapr-pubsub-manual.bicep [hunk 9]
```diff
@@ -20,7 +20,7 @@ resource publisher 'Applications.Core/containers@2023-10-01-preview' = {
       }
     }
     container: {
-      image: 'radius.azurecr.io/magpie:latest'
+      image: 'radius.ghcr.io/magpie:latest'
```
Comment (jasonviviano) [inline docs/content/reference/resource-schema/dapr-schema/dapr-pubsub/snippets/dapr-pubsub-manual.bicep]: TODO: Same as here

File: docs/content/reference/resource-schema/dapr-schema/dapr-statestore/snippets/dapr-statestore-recipe.bicep [hunk 10]
```diff
@@ -11,7 +11,7 @@ resource myapp 'Applications.Core/containers@2023-10-01-preview' = {
   properties: {
     application: application
     container: {
-      image: 'radius.azurecr.io/magpie:latest'
+      image: 'radius.ghcr.io/magpie:latest'
```
Comment (jasonviviano) [inline docs/content/reference/resource-schema/dapr-schema/dapr-statestore/snippets/dapr-statestore-recipe.bicep]: TODO: Same as here

File: docs/content/guides/recipes/howto-author-recipes/index.md (hunk 1 above)
Comment (AaronCrawfis) [reply docs/content/guides/recipes/howto-author-recipes/index.md]:
```suggestion
rad bicep publish --file myrecipe.bicep --target br:ghcr.io/USERNAME/recipes/myrecipe:1.1.0
```

File: docs/content/guides/recipes/howto-author-recipes/index.md (hunk 2 above)
Comment (AaronCrawfis) [reply docs/content/guides/recipes/howto-author-recipes/index.md]:
```suggestion
rad recipe register myrecipe --environment myenv --resource-type Applications.Datastores/redisCaches --template-kind bicep --template-path ghcr.io/USERNAME/recipes/myrecipe:1.1.0
```

File: docs/content/guides/recipes/howto-author-recipes/snippets/environment.bicep (hunk 3 above)
Comment (AaronCrawfis) [reply docs/content/guides/recipes/howto-author-recipes/snippets/environment.bicep]:
```suggestion
          templatePath: 'https://ghcr.io/USERNAME/recipes/myrecipe:1.1.0'
```

File: docs/content/reference/resource-schema/core-schema/container-schema/_index.md (hunk 4 above)
Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/container-schema/_index.md]:
```suggestion
| image | y | The registry and image to download and run in your container. Follows the format `<registry-hostname>:<port>/<image-name>:<tag>` where registry hostname is optional and defaults to the Docker public registry, port is optional and defaults to 443, tag is optional and defaults to `latest`.| `ghcr.io/USERNAME/myimage:latest`
```

File: docs/content/reference/resource-schema/core-schema/environment-schema/index.md (hunk 6 above)
Comment (AaronCrawfis) [reply docs/content/reference/resource-schema/core-schema/environment-schema/index.md]:
```suggestion
| templatePath | y | The path to the Recipe contents. For Bicep Recipes this is a Bicep module registry address. | `'ghcr.io/USERNAME/recipes/myrecipe:1.0'`
```

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**